### PR TITLE
feat(DSC): support a new resource to manage DSC

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dsc' TESTARGS='-run=TestAccDscInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dsc -v -run=TestAccDscInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccDscInstance_basic
=== PAUSE TestAccDscInstance_basic
=== CONT  TestAccDscInstance_basic
--- PASS: TestAccDscInstance_basic (83.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dsc       83.728s
```

### CODE REVIEW

File: docs/resources/dsc_instance.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+## Import
+
+The dsc instance can be imported using the `id`, e.g.
+
+```
+$ terraform import huaweicloud_dsc_instance.test 0ce123456a00f2591fabc00385ff1234
```
Comment (houpeng80) [inline docs/resources/dsc_instance.md:85]: the id format is not correct

File: docs/resources/dsc_instance.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+```HCL
+  resource "huaweicloud_dsc_instance" "test" {
+    edition                    = "base_standard"
+    charging_mode              = "prePaid"
+    period_unit                = "month"
+    period                     = 1
+    auto_renew                 = false
+    obs_expansion_package      = 1
+    database_expansion_package = 1
+  }
```
Comment (Lance52259) [inline docs/resources/dsc_instance.md]:
```suggestion
resource "huaweicloud_dsc_instance" "test" {
  edition                    = "base_standard"
  charging_mode              = "prePaid"
  period_unit                = "month"
  period                     = 1
  auto_renew                 = false
  obs_expansion_package      = 1
  database_expansion_package = 1
}
```

Comment (hmbig2) [reply docs/resources/dsc_instance.md]: done

File: docs/resources/dsc_instance.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/dsc_instance.md:85]: It's correct

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+				ForceNew: true,
+			},
+			"edition": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The edition of DSC.`,
+				ValidateFunc: validation.StringInSlice([]string{
+					"base_standard", "base_professional",
+				}, false),
```
Comment (Lance52259) [inline huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]:
```suggestion
				ValidateFunc: validation.StringInSlice([]string{
					resourceSpecCodeStandardBase, resourceSpecCodeProBase,
				}, false),
```

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+		Target:       []string{"COMPLETE"},
+		Refresh:      waitForBmsInstanceDelete(ctx, d, meta),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        20 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.DiagErrorf("Error deleting DSC instance: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]:
```suggestion
		return diag.Errorf("Error deleting DSC instance: %s", err)
```

please does not use **fmtp** package in new resource

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+				"resourceSpecCode":      resourceSpecCodeStandardObs,
+				"resourceSize":          utils.ValueIngoreEmpty(size),
+				"resourceSizeMeasureId": 47,
+			})
+		}
+
+		if size, ok := d.GetOk("database_expansion_package"); ok {
+			rst = append(rst, map[string]interface{}{
+				"cloudServiceType":      cloudServiceType,
+				"productId":             "OFFI571544962480533516",
```
Comment (ShiChangkuo) [inline huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]: **OFFI571544962480533516** is magic string, you can define it with a more meaningful variable.

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]: done

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]: done

**Pull request merged**